### PR TITLE
Fixing attention mask & cache

### PR DESCRIPTION
**Pull request by NohTow**
Hello,

Following https://github.com/OFA-Sys/OFA/pull/214#issuecomment-1233919521 and https://github.com/OFA-Sys/OFA/pull/222#event-7310098999, I found another place where the attention mask is wrong (or at least not in the HF format):
```
 if decoder_input_ids.eq(self.config.pad_token_id).any():
            attention_mask = decoder_input_ids.eq(self.padding_idx)
```
Commenting these lines is fine since the attention mask should be fed to the model in the first place. The issue seems to be that this line returns a boolean mask instead of 0s and 1s. Maybe casting it can also works, but I do not think this is good practice to overwrite the attention mask computed before hand. Maybe computing it only if there is none given to the model.

Also, as stated here: https://github.com/OFA-Sys/OFA/issues/171#issuecomment-1245401386, using cache (`use_cache=True`) results in bad (or at least worse) outputs.
I digged into that and found the reason ; while the states are well cached, the `attn_bias` is not the same when using or not the cache since it is computed like so: `self_attn_bias += self.get_rel_pos_bias(all_prev_output_tokens, idx).unsqueeze(0)` which rely on `all_prev_output_tokens`, which, when using cache, is only the last token (when using cache, we only feed the last token as input to the model).
For a quick fix, I changed `prepare_inputs_for_generation` so we have every token as inputs (only the last one is fed to the decoder, since the same line is in the its forward() function. Also adapting the encoder attention mask in case past is used (since it is before processing the input ids).

I take this opportunity to ask again: @JustinLin610, do you plan to release fine-tuned checkpoint for tiny and medium model ? Would be cool for people that want to further train these models and do not have access to a lot of GPUs.

### CODE REVIEW

**Comment by JustinLin610** [comment]
Good! Seems quite reasonable. Would you provide a colab notebook for me to test if everything works fine so that I can directly merge it. Sorry for not having unit tests. As to the finetuned checkpoints, I think that would be fine for me to release them. Probably next month when I got more free time to double check the quality of the models.

**Review by JustinLin610 — APPROVED** [review]
